R/iterate-helpers.R

```diff
@@ -0,0 +1,115 @@
+#' Iteration helpers
+#'
+#' @description
+#' These functions are suitable for use with the `next_req` argument to
+#' [req_perform_iteratively()]. Each implements iteration for a common
+#' pagination pattern:
+#'
+#' * `iterate_with_offset()` increments a query parameter, e.g. `?page=1`,
+#'   `?page=2`, or `?offset=1`, `offset=21`.
+#' * `iterate_with_cursor()` updates a query parameter with the value of a
+#'    cursor found somewhere in the response.
```

Comment:
Restricting `iterate_with_offset()` and `iterate_with_cursor()` to a query parameter makes the implementation much simpler. I hope this covers most APIs but it definitely doesn't cover the main one I have to use...

Reply:
I think it's a reasonable trade-off. My plan is to eventually have vignette that describes how to handle more complex situations.